app/main.py

```diff
@@ -1,3 +1,18 @@
-def cache(func):
-    # Write your code here
-    pass
+from functools import wraps
+from typing import Any, Tuple, Dict, Callable
+
+
+def cache(func: Callable) -> Callable:
+    results: Dict[Tuple[Any, ...], Any] = {}
+
+    @wraps(func)
+    def wrapper(*args: Any, **kwargs: Any) -> Any:
+        arguments = args + tuple(kwargs.items())
+        if arguments in results:
+            print("Getting from cache")
+            return results[arguments]
```

Comment:
and delete this

Reply:
If I delete this part of the code, how will he know what arguments is?